### PR TITLE
chore: Combine review and end nodes

### PR DESCRIPTION
**Pull request by timarney**
# Summary | Résumé

fixes: https://github.com/cds-snc/platform-forms-client/issues/4217

- Update to not display review node as a standalone node

- Adds custom end nodes

<hr>

**End when a review page is showing:**

Screenshot 2024-08-27 at 7 44 04 AM

**End when no review page:**

end-when-no-review

## Testing

Note for testing you should test with a single page and a multi-page form.

### CODE REVIEW

**Comment by timarney** [comment]
@anikbrazeau PR is updated to make the end node clickable 

We'll need to figure out what content should be in the panel.

File: i18n/translations/en/common.json [hunk 1]
```diff
@@ -159,7 +159,7 @@
   "logic": {
     "start": "Start page",
     "end": "Confirmation page",
-    "review": "Review page",
+    "review": "End (Review and confirm)",
```
Comment (anikbrazeau) [inline i18n/translations/en/common.json:162]:
```suggestion
    "review": "End (Review page and Confirmation)",
```

File: i18n/translations/fr/common.json [hunk 2]
```diff
@@ -160,7 +160,7 @@
   "logic": {
     "start": "Page de départ",
     "end": "Page de confirmation",
-    "review": "Page récapitulative",
+    "review": "End (Review and confirm) [fr]",
```
Comment (anikbrazeau) [inline i18n/translations/fr/common.json:163]:
```suggestion
    "review": "Fin (Page récapitulative et confirmation)",
```

File: i18n/translations/fr/form-builder.json [hunk 3]
```diff
@@ -925,6 +925,14 @@
...
       "option": "Pour faire emprunter un chemin précis en fonction d'une réponse choisie."
     },
+    "endNode": {
+      "label": "End",
+      "confirmContent": "Confirmation message"
+    },
+    "endNodeWithReview": {
+      "label": "End (Review and confirm)",
+      "reviewContent": "Review"
+    },
```
Comment (anikbrazeau) [inline i18n/translations/fr/form-builder.json:935]:
```suggestion
    "endNode": {
      "label": "Fin",
      "confirmContent": "Page de confirmation"
    },
    "endNodeWithReview": {
      "label": "Fin (Page récapitulative et confirmation)",
      "reviewContent": "Page récapitulative"
    },
```

File: i18n/translations/en/form-builder.json [hunk 4]
```diff
@@ -925,6 +925,14 @@
...
       "option": "To take a specific path depending on a selected option."
     },
+    "endNode": {
+      "label": "End",
+      "confirmContent": "Confirmation message"
+    },
+    "endNodeWithReview": {
+      "label": "End (Review and confirm)",
+      "reviewContent": "Review"
+    },
```
Comment (anikbrazeau) [inline i18n/translations/en/form-builder.json:935]:
```suggestion
    "endNode": {
      "label": "End",
      "confirmContent": "Confirmation page"
    },
    "endNodeWithReview": {
      "label": "End (Review and Confirmation)",
      "reviewContent": "Review page"
    },
```

**Comment by timarney** [comment]
@anikbrazeau 

I pushed some updates to add a spot where you can add a description and or we can leave it blank.

I also updated to "Section name" / "Page name" to use the (Review and Confirm) label if a review exists --- if not it will just say "End"

Screenshot 2024-08-29 at 8 54 41 AM

**Review by thiessenp-cds — APPROVED** [review]
In the future we may want to think about a way to allow an element to be focussed on the Review/end node. This way a user tabbing through logical view elements will have it read out. 
This behavior is the same as on Staging. So more of an afterthought of the current behavior.
Nice work! LGTM :)

File: i18n/translations/en/form-builder.json [hunk 5]
```diff
@@ -925,6 +925,14 @@
       "page": "To change how pages are ordered and connected.",
       "option": "To take a specific path depending on a selected option."
     },
+    "endNode": {
+      "label": "End",
```
Comment (anikbrazeau) [inline i18n/translations/en/form-builder.json:929]:
```suggestion
      "label": "End of form",
```

File: i18n/translations/en/form-builder.json [hunk 6]
```diff
@@ -961,6 +969,9 @@
       "text1": "Add a <strong>New page</strong> from the",
       "text2": "Pages tab",
       "text3": "before adding a branch."
+    },
+    "endPage": {
+      "description": "Select another node to add rules, you cannot add rules to the End"
```
Comment (anikbrazeau) [inline i18n/translations/en/form-builder.json:974]:
```suggestion
      "description": "Select another page. You cannot add rules to the end of the form."
```

File: i18n/translations/fr/form-builder.json [hunk 7]
```diff
@@ -925,6 +925,14 @@
       "page": "Pour modifier la façon dont les pages sont ordonnées et connectées.",
       "option": "Pour faire emprunter un chemin précis en fonction d'une réponse choisie."
     },
+    "endNode": {
+      "label": "Fin",
```
Comment (anikbrazeau) [inline i18n/translations/fr/form-builder.json:929]:
```suggestion
      "label": "Fin du formulaire",
```

File: i18n/translations/fr/form-builder.json [hunk 8]
```diff
@@ -961,6 +969,9 @@
       "text1": "Ajoutez une <strong>Nouvelle page</strong> dans",
       "text2": "l'onglet Pages",
       "text3": "avant d'ajouter un embranchement."
+    },
+    "endPage": {
+      "description": "Select another node to add rules, you cannot add rules to the End"
```
Comment (anikbrazeau) [inline i18n/translations/fr/form-builder.json:974]:
```suggestion
      "description": "Sélectionnez une autre page. Vous ne pouvez pas ajouter de règles à fin du formulaire."
```

**Review by thiessenp-cds — APPROVED** [review]

**Pull request merged**